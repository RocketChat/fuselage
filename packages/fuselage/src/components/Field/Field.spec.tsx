import React from 'react';
import ReactDOM from 'react-dom';

import { Field } from '../..';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Field />, div);
  ReactDOM.unmountComponentAtNode(div);
});

describe('Label', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Field.Label />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Description', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Field.Description />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Row', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Field.Row />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Error', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Field.Error />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe('Hint', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Field.Hint />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});
