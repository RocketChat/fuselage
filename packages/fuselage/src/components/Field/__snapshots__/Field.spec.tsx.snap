// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Field Component] renders WithCheckbox without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <label
          class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
          for="fieldWithCheckbox"
        >
          Label
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
              id="fieldWithCheckboxInfo"
            >
              this is a info label
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="this is a info label"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </label>
        <label
          class="rcx-box rcx-box--full rcx-check-box"
        >
          <input
            aria-describedby="fieldWithCheckboxInfo"
            class="rcx-box rcx-box--full rcx-check-box__input"
            id="fieldWithCheckbox"
            type="checkbox"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-check-box__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
      >
        Description
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
      >
        Error feedback
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
        >
          Hint
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Field Component] renders WithRadioButton without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <label
          class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
          for="fieldWithRadio"
        >
          Label
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
              id="fieldWithRadioInfo"
            >
              this is a info label
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="this is a info label"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </label>
        <label
          class="rcx-box rcx-box--full rcx-radio-button"
        >
          <input
            aria-describedby="fieldWithRadioInfo"
            class="rcx-box rcx-box--full rcx-radio-button__input"
            id="fieldWithRadio"
            type="radio"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-radio-button__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
      >
        Description
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
      >
        Error feedback
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
        >
          Hint
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Field Component] renders WithTextArea without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
        for="fieldWithTextArea"
      >
        Label
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
            id="fieldWithTextAreaInfo"
          >
            this is a info label
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="this is a info label"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
      >
        Description
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <textarea
          aria-describedby="fieldWithTextAreaInfo"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-textarea rcx-input-box"
          cols="1"
          id="fieldWithTextArea"
          size="1"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
      >
        Error feedback
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
        >
          Hint
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Field Component] renders WithTextInput without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <label
        class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
        for="fieldWithText"
      >
        Label
        <span
          class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
        >
          <span
            hidden=""
            id="fieldWithTextInfo"
          >
            this is a info label
          </span>
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
            title="this is a info label"
          >
            
          </i>
        </span>
        <span
          aria-hidden="true"
          class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
        >
          *
        </span>
      </label>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
      >
        Description
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <input
          aria-describedby="fieldWithTextInfo"
          class="rcx-box rcx-box--full rcx-box--animated rcx-input-box--type-text rcx-input-box"
          id="fieldWithText"
          size="1"
          type="text"
        />
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
      >
        Error feedback
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
        >
          Hint
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link
        </a>
      </span>
    </div>
  </div>
</body>
`;

exports[`[Field Component] renders WithToggleSwitch without crashing 1`] = `
<body>
  <div>
    <div
      class="rcx-box rcx-box--full rcx-field"
    >
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <label
          class="rcx-box rcx-box--full rcx-label  rcx-box rcx-box--full rcx-field__label"
          for="fieldWithToggle"
        >
          Label
          <span
            class="rcx-box rcx-box--full rcx-label__info rcx-css-wcp0mp"
          >
            <span
              hidden=""
              id="fieldWithToggleInfo"
            >
              this is a info label
            </span>
            <i
              aria-hidden="true"
              class="rcx-box rcx-box--full rcx-icon--name-info-circled rcx-icon"
              title="this is a info label"
            >
              
            </i>
          </span>
          <span
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-label__required rcx-css-b7g1a9"
          >
            *
          </span>
        </label>
        <label
          class="rcx-box rcx-box--full rcx-toggle-switch"
        >
          <input
            aria-describedby="fieldWithToggleInfo"
            class="rcx-box rcx-box--full rcx-toggle-switch__input"
            id="fieldWithToggle"
            type="checkbox"
          />
          <i
            aria-hidden="true"
            class="rcx-box rcx-box--full rcx-toggle-switch__fake"
          />
        </label>
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__description"
      >
        Description
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__error"
      >
        Error feedback
      </span>
      <span
        class="rcx-box rcx-box--full rcx-field__row"
      >
        <span
          class="rcx-box rcx-box--full rcx-field__hint"
        >
          Hint
        </span>
        <a
          class="rcx-box rcx-box--full rcx-field__link"
          href="#"
          target="_blank"
        >
          Link
        </a>
      </span>
    </div>
  </div>
</body>
`;
